src/mpl/src/gpu/mpl_gpu_cuda.c

```diff
@@ -196,9 +196,9 @@ int MPL_gpu_free(void *ptr)
     return MPL_ERR_GPU_INTERNAL;
 }
 
-int MPL_gpu_init(int *device_count, int *max_dev_id_ptr)
+int MPL_gpu_init()
 {
-    int count, max_dev_id = -1;
+    int count, tmp_max_dev_id = -1;
```

Comment:
I don't see the point of using temp variables here. Just use the global variables directly.